### DISCUSSION TITLE
Add isRefreshing flag

### DISCUSSION THREAD

**Discussion by matheusmb** [TanStack/query#2459] [Ideas]
It would by nice to have an `isRefreshing`  or `isRefetching` flag to indicate that a refetch is being performed. In my project I created the following custom hook that enhances useQuery with this feature:

```ts
export function useEnhancedQuery<
  TQueryFnData = unknown,
  TError = unknown,
  TData = TQueryFnData,
  TQueryKey extends QueryKey = QueryKey,
>(
  queryKey: TQueryKey,
  queryFn: QueryFunction<TQueryFnData, TQueryKey>,
  options?: UseQueryOptions<TQueryFnData, TError, TData, TQueryKey>,
) {
  const queryResult = useQuery(queryKey, queryFn, options);
  const isRefreshing = !queryResult.isLoading && queryResult.isFetching;
  return {
    ...queryResult,
    isRefreshing,
  };
}
```

I use this flag to display a refreshing indicator as feedback. I think this feature can be useful for other users.

Thank you.

**Comment by TkDodo** [comment]
I added a PR: https://github.com/tannerlinsley/react-query/pull/2469

but I have some concerns about `notifyOnChangePropsExclusion` (see the PR for more details pls)
